Sources/ZcashLightClientKit/Block/Enhance/BlockEnhancer.swift

```diff
@@ -92,15 +92,17 @@ extension BlockEnhancerImpl: BlockEnhancer {
                     do {
                         switch transactionDataRequest {
                         case .getStatus(let txId):
-                            let response = try await blockDownloaderService.fetchTransaction(txId: txId.data)
+                            // ServiceMode to resolve
+                            let response = try await blockDownloaderService.fetchTransaction(txId: txId.data, mode: .defaultTor)
```

Comment:
Fetches of a transaction should run in their own fetch group:
```suggestion
                            let response = try await blockDownloaderService.fetchTransaction(
                                txId: txId.data,
                                mode: ServiceMode.txIdGroup(prefix: "fetch", txId: txId.data)
                            )
```

Then we need to disconnect these fetched from the sync engine. Note that `BlockEnhancer` takes a `range` but does nothing with it, instead polling `transactionDataRequests()` to decide what to do. The `EnhanceAction` should be removed entirely from `cbp_state_machine`, and instead `BlockEnhancer` should be periodically run (however is best: this could be a long-running task that wakes up every e.g. 10 seconds to check for changes).